torch/csrc/jit/compiler/src/llvm_codegen.cc

```diff
@@ -17,16 +22,46 @@ LLVMCodeGen::LLVMCodeGen() : irb_(context_) {
   module_->setTargetTriple(
       jit_->getTargetMachine().getTargetTriple().normalize());
 
-  // Emit prototype.
   int32Ty_ = llvm::Type::getInt32Ty(context_);
 
-  llvm::FunctionType* fntype = llvm::FunctionType::get(int32Ty_, {}, false);
+  // Emit prototype.
+  std::vector<llvm::Type *> params;
+  for (int i = 0; i < args.size(); i++) {
+    params.push_back(llvm::Type::getInt32PtrTy(context_));
+    varToArg_[args[i]->data().node()] = i;
+  }
+  llvm::FunctionType* fntype = llvm::FunctionType::get(int32Ty_, params, false);
   fn_ = llvm::Function::Create(
-      fntype, llvm::Function::ExternalLinkage, "pytorch", module_.get());
+    fntype, llvm::Function::PrivateLinkage, "pytorch", module_.get());
+  for (int i = 0; i < args.size(); i++) {
+    fn_->addParamAttr(i, llvm::Attribute::NoAlias);
+  }
+
+  // Emit wrapper to unpack argument vector.
+  auto i32pp = int32Ty_->getPointerTo()->getPointerTo();
```

Comment:
Minor nitpicking: In general, I am not a big fan with auto. If the real type is super long, maybe a typedef? With the real type, it is much easier for reading, although writing is a bit more work.

Reply:
In the interest of time I'm going to land with `auto`s but I'm not wedded to them :-).